### DISCUSSION TITLE
barrier=bollard

### DISCUSSION THREAD

**Discussion by bwbuz** [streetcomplete/StreetComplete#6000] [Q&A]
Hello, I couldn't find a way to add a newly discovered bollard today. I was looking in things, among other things. Does anyone know how this works or is it missing here? Kind regards

**Comment by HolgerJeromin** [comment]
With StreetComplete this is not possible. 
With the "expert" fork SCEE it is in a simple form: https://github.com/Helium314/SCEE/issues/414

**Reply by matkoniecz** [reply]
is it allowing to attach them to roads? to mark road as blocked by bollard? or do you need to do this manually in some editor?

**Reply by HolgerJeromin** [reply]
The bollard is snapping to the road.
Quite easy with good visual feedback

**Reply by westnordost** [reply]
Hmm, how is this done? Can you add any nodes to ways, or just those that have to be on ways?

If it is the latter, i.e. is safe to use, I don't see a reason why it couldn't be part of StreetComplete proper.

**Reply by mnalis** [reply]
> Hmm, how is this done? Can you add any nodes to ways, or just those that have to be on ways?

There is special long-press functionality called _"Insert node into way"_.

You first select the way and position on it, and then you can pick any POI from id-tagging-schema to be inserted into that way, e.g. `amenity=post_box` on `building=yes` closed way, or `barrier=bollard` on `highway=residential` way, but also insensible ones like `natural=tree` on `highway=motorway`.

So I'd say it would need specific filtering (depending which nodes are allowed on which ways) to be safe for StreetComplete. Looks like this:


![small_Screenshot_20241121_021241_SCEE](https://github.com/user-attachments/assets/3794ad3d-f108-4460-827a-b31546cae9aa) ![small_Screenshot_20241121_021333_SCEE](https://github.com/user-attachments/assets/6272e189-4f14-44a6-9fec-50aade87be06)

**Reply by westnordost** [reply]
Oh, I see. Well, I would implement it as such as that you can select the preset right from the start like any other, but presets that should only be vertices of certain features can only be placed snapping to the lines/outlines of such features (like when putting a housenumber on a building outline as an entrance).
This requires additional  meta information that isn't available in the id editor presets, as far as I know. E.g. it shouldn't be possible to add e.g. speed cushions on building outlines, but *just* on roads. Most features shouldn't snap on lines at all, e.g. ATMs (yes, I know this is a common practice).

**Comment by matkoniecz** [comment]
Bollards were deliberately skipped (at least for now) from Things layer.

Some should be marked as attached to footways/roads/cycleways and handling this was too tricky, freefloating ones are not so important to map like manholes.

**Reply by mnalis** [reply]
- See also https://github.com/streetcomplete/StreetComplete/discussions/5752 and [#5022 (comment)](https://github.com/streetcomplete/StreetComplete/issues/5022#issuecomment-2081189790)
